### PR TITLE
feat/UNTIL-6762: list all and one order endpoints

### PR DESCRIPTION
**Pull request by desicne**

### CODE REVIEW

File: src/v2/orders.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+export interface OrdersResponse {
+    data: Order[]
+    metadata: Record<string, unknown>
+    next?: () => Promise<OrdersResponse>
+}
+
+export interface OrdersQuery {
+    limit?: number
+    uri?: string
+    query?: {
```
Comment (maurogri-th) [inline src/v2/orders.ts:21]: The query object seems to be missing most of it. HEre is the link to the filters that are expected
https://unz.atlassian.net/wiki/spaces/TA/pages/1159004297/UFC+Order+Management+Tech+Spec#3.1.4-Orders-API

File: src/v2/orders.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        active?: boolean
+        extended?: boolean
+        location?: string
+    }
+}
+
+declare type RecurringFormatType = 'scheduled' | 'unscheduled' | 'oneclick'
+declare type ReturnFormatType = 'refundable' | 'capturable' | 'cancellable'
+
+export interface Order {
```
Comment (maurogri-th) [inline src/v2/orders.ts:32]: This is the basic info of an order, that would make sense for the listing. But an order is made of a lot of different components, like transaction, payments, etc. Adding here the doc:
https://unz.atlassian.net/wiki/spaces/TA/pages/1139474576/Order+data+model+detailed+draft

**Review by maurogri-th — COMMENTED** [review]
The order query needs to be extended and the order detail has to have the inner objects defined for the get.

File: src/v2/orders.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    id?: string
+    orderStatus?: string
+    totalAmount?: number | null
+    currency?: string | null
+    salesChannel?: string | null
+    origin?: string | null
+    recurring?: RecurringFormatType | null
+    returnable?: ReturnFormatType | null
```
Comment (maurogri-th) [inline src/v2/orders.ts:40]: Where is this one coming from?

File: src/v2/orders.ts (hunk 2 above)
Comment (desicne) [reply src/v2/orders.ts:32]: Ok, stupid of me.. I only thought it was order part that was there.

File: src/v2/orders.ts (hunk 1 above)
Comment (desicne) [reply src/v2/orders.ts]: Think now it is good

**Review by maurogri-th — APPROVED** [review]

**Pull request merged**